summernote/summernote
# Added container option for global elements added by summernote and Bootstrap

#### What does this PR do?

Allows for custom containers to be specified upon initialization of the editor, such as:

```
<div class="summernote"></div>
<div id="my-container"></div>
<script type="text/javascript">
    $('.summernote').summernote({
        container: '#my-container'
    });
</script>
```

As it stands now, `'body'` is hardcoded in multiple places as  the default place for global DOM insertions. This causes issues when custom CSS rules need to be applied for specific summernote tooltips, popovers, and palettes. This allows for the specific insertion selector to be changed, with the default options being `'body'`.
#### Where should the reviewer start?
- start on the src/js/bs3/settings.js file, then the rest of the modified src/js/bs3/*.js files
#### How should this be manually tested?
- Using the snippet provided above
- Check to make sure that the UI elements are inserted into the correct place when fully loaded
- All functionality should remain the same
#### Any background context you want to provide?

The main reason I've added this is because of CSS framework conflicts. Bootstrap is a requirement for summernote, but our sites legacy CSS has conflicts with it. As a work around, I've created a "sandboxed" version of Bootstrap to be applied only on areas where summernote is to be used, which fixes the primary editor. But the global UI elements were unstyled, this allows for those "sandbox" CSS rules to be applied by a custom element.
#### Relevant Issues
#1963
#### Screenshots (if for frontend)

![capture](https://cloud.githubusercontent.com/assets/4315904/17946027/1fb3a7ae-6a0c-11e6-9e1b-8786b2f2c524.PNG)
